Add 0len ext to report (#43)
wangshan/draft-wang-ppm-dap-taskprov: draft-wang-ppm-dap-taskprov.md

The definition of `Time`, `Duration`, `Url`, and `QueryType` follow those in
{{!DAP}}.

> 

Reviewer: Move this new text to the "Client Behavior" section.

> In addition, a new `task_prov` extension is offered in both Leader and Helper's

Reviewer: To align a bit better with the header.
```suggestion
In addition, a new `taskprov` extension is offered in both Leader and Helper's
```

> OPEN ISSUE: In case of opt-out, would it be useful to specify how to report
> this to the Author?

Finally, the Helper completes the aggregate initialize request as usual,
deriving the VDAF verification key for the task as described in
{{vdaf-verify-key}}.

> During Helper aggregate initialization, if Helper's report share does not
> present a `taskprov` extension type, Helper MUST abort the aggregate request
> with "invalidMessage".

Reviewer: ```suggestion
During Helper aggregate initialization, if any Helper's report share does not
include the `taskprov` extension with an empty payload, then the Helper MUST
abort the aggregate request with "invalidMessage".
```

> OPEN ISSUE: In case of opt-out, would it be useful to specify how to report
> this to the Author?

Finally, the Helper completes the aggregate initialize request as usual,
deriving the VDAF verification key for the task as described in

> {{vdaf-verify-key}}. During the process, if Helper's report share does not

Reviewer: This should be a separate paragraph.

present a `task_prov` extension type, Helper SHOULD abort the aggregate request

> with "invalidMessage".

Reviewer: What's the intent here? To ensure that all Clients indicate support for `task_prov`? I think this should be a MUST.

Author: The intent is to keep consistency, if the client opt in to taskprov, it should set both the ext type and dap-taskprov header. agree with the must

and Aggregators have an inconsistent view of the the task configuration, then
aggregation of reports will fail. This is guaranteed by the binding of the task
ID (derived from the task configuration) to report shares provided by HPKE

> encryption. Furthermore, the presence of `task_prov` extension type in report

Reviewer: ```suggestion
encryption. Furthermore, the presence of `task_prov` extension type in the report
```

share means Aggregators that do not recognize Taskprov extension must abort with
`invalidMessage`, as described in ({{Section 4.4.3 of !DAP}}). This prevents a

> malicious Leader from provisioning a modified task in Helper with other means,

Reviewer: ```suggestion
malicious Author from provisioning a modified task to each party by some other other means,
```

> which can lead to compromised privacy guarantee in aggregation results.

Reviewer: * "aggregation results" is not a term we're using. Perhaps you mean "aggregate result"?
* Is privacy the only risk?

Author: I can't think of what else a malicious author can achieve with this particular risk
